### PR TITLE
Add helper for locking post saving

### PR DESCRIPTION
**Pull request by mbmjertan**
Adds three helpers for locking post saving:
* `lockPostEditing`, locking a post save with a lock handle determined from a block client ID and attribute key
* `unlockPostEditing`, removing a lock created with `lockPostEditing` by removing the lock with the lock handle matching a client ID and attribute key
* `lockIfUndefined`, locking and unlocking a post save automatically if the attribute value is undefined, given a block client ID, attribute key and value

Usage example:
```
			<RichText
				className={headingClass}
				placeholder={placeholder}
				value={headingContent}
				onChange={(value) => { setAttributes({ [getAttrKey('headingContent', attributes, manifest)]: value }); lockIfUndefined(clientId, getAttrKey('headingContent', attributes, manifest), value); }}
				allowedFormats={[]}
				data-id={unique}
			/>
```

I'll do a PR to eightshift-docs documenting this helper as well.

### CODE REVIEW

File: scripts/helpers/lock-editing.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * A lock is created under the name undefined-lock-[blockClientId]-[attributeKey],
+ * meaning that a lock can be created for each attribute key of each block.
+ * Multiple locks can be present and all of them have to be removed before saving is unlocked.
+ *
+ * @param {string} blockClientId A block's client ID
+ * @param {string} attributeKey The attribute key
+ */
+
+export const lockIfUndefined = (blockClientId, attributeKey, value) => {
+  if (typeof value === 'undefined' || value === null || value === '') {
```
Comment (goranalkovic-infinum) [inline scripts/helpers/lock-editing.js:41]: Maybe we shouldn't equate `undefined` and `null` here

**Review by dadadavorin — DISMISSED** [review]
👍  Just solve the linter issues 😄

**Comment by mbmjertan** [comment]
> 👍 Just solve the linter issues 😄

Seems like `npm install` is crashing. Will look into it.

File: scripts/helpers/index.js [hunk 2]
```diff
@@ -19,3 +19,8 @@ export {
 	truncateMiddle,
 	unescapeHTML
 } from './text-helpers';
+export {
+	lockPostEditing,
+	unlockPostEditing,
+	lockIfUndefined
+} from './lock-editing';
```
Comment (iruzevic) [inline scripts/helpers/index.js:26]: this is specific to the editor only not the frontend so please put this in the editor folder not helpers folder and also export order should be alphabetical order. tnx

File: scripts/helpers/lock-editing.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/**
```
Comment (iruzevic) [inline scripts/helpers/lock-editing.js:1]: please with for all functions

return type, usage and output

like in other helpers
https://github.com/infinum/eightshift-frontend-libs/blob/develop/scripts/editor/props.js

**Review by iruzevic — CHANGES_REQUESTED** [review]
a few small corrections but great job

File: scripts/helpers/lock-editing.js (hunk 1 above)
Comment (mbmjertan) [reply scripts/helpers/lock-editing.js:41]: The idea was to lock post saving if there's an attribute that's missing, and I'd wager `null` and an empty string fall under that. I'm not sure what you're suggesting, but I'd suggest using the `lockPostEditing` helper for different use cases.

**Review by goranalkovic-infinum — DISMISSED** [review]

File: scripts/editor/index.js [hunk 4]
```diff
@@ -13,6 +13,11 @@ export {
 } from './get-options';
 export { getPaletteColors } from './get-palette-colors';
 export { inserter } from './inserter';
+export {
+	lockIfUndefined,
+	lockPostEditing,
+	unlockPostEditing
```
Comment (volfkarlo) [inline scripts/editor/index.js:19]:
```suggestion
	unlockPostEditing,
```

File: scripts/editor/lock-editing.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ * ```
+ */
+
+export const lockIfUndefined = (blockClientId, attributeKey, value) => {
+  if (typeof value === 'undefined' || value === null || value === '') {
+    lockPostEditing(blockClientId, attributeKey);
+    return;
+  }
+  unlockPostEditing(blockClientId, attributeKey);
+};
```
Comment (volfkarlo) [inline scripts/editor/lock-editing.js:70]: New row at the end.
```suggestion
};

```

**Review by volfkarlo — APPROVED** [review]

**Review by dingo-d — DISMISSED** [review]
Looks good to me, just solve Karlo's comments and that's it 😄

**Review by goranalkovic-infinum — APPROVED** [review]

**Review by dingo-d — APPROVED** [review]

**Review by dadadavorin — APPROVED** [review]

**Review by iruzevic — APPROVED** [review]

**Pull request merged**